expr.go

```diff
@@ -293,23 +319,40 @@ func getStringArg(e *expr, n int) (string, error) {
 		return "", ErrMissingArgument
 	}
 
-	if e.args[n].etype != etString {
-		return "", ErrBadType
+	return doGetStringArgDefault(e.args[n], "")
+}
+
+func getStringArgDefault(e *expr, n int, s string) (string, error) {
+	if len(e.args) <= n {
+		return s, nil
 	}
 
-	return e.args[n].valStr, nil
+	return doGetStringArgDefault(e.args[n], s)
 }
 
-func getStringArgDefault(e *expr, n int, s string) (string, error) {
+func getStringNamedOrPosArgDefault(e *expr, k string, n int, s string) (string, error) {
 	if len(e.args) <= n {
```

Comment:
It's a bit of an edge case but;
In functions with >1 named/optional argument this check may be invalid when one of the "interior" arguments are dropped. Examples of such functions:

- `summarize(seriesList, intervalString, func='sum', alignToFrom=False)`

Functions we have that could be affected:
- `threshold(value, label=None, color=None)` (not supported here yet - but I've implemented privately)
- `timeShift(seriesList, timeShift, resetEnd=True, alignDST=False)` (we don't support alignDST yet)
- `transformNull(seriesList, default=0, referenceSeries=None)` (we don't support referenceSeries yet)

Other graphite functions not supported here yet:
- `linearRegression(seriesList, startSourceAt=None, endSourceAt=None)`
- `sinFunction(name, amplitude=1, step=60)`
- `smartSummarize(seriesList, intervalString, func='sum', alignToFrom=False)`
- `substr(seriesList, start=0, stop=0)`